editor/src/components/canvas/canvas-strategies/strategies/absolute-duplicate-strategy.spec.browser2.tsx

```diff
@@ -93,16 +101,91 @@ describe('Absolute Duplicate Strategy', () => {
         <div style={{ width: '100%', height: '100%' }} data-uid='aaa'>
           <div
             style={{ backgroundColor: '#aaaaaa33', position: 'absolute', left: 40, top: 50, width: 200, height: 120 }}
-            data-uid='bbb'
+            data-uid='hello'
             data-testid='bbb'
-          />
-          <div
+            />
+            <div
             style={{ backgroundColor: '#aaaaaa33', position: 'absolute', left: 80, top: 25, width: 200, height: 120 }}
-            data-uid='hello'
+            data-uid='bbb'
             data-testid='bbb'
           />
         </div>
       `),
     )
   })
+  describe('with fragments', () => {
+    setFeatureForBrowserTests('Fragment support', true)
+    it('duplicates the selected absolute element when pressing alt, even if it is a fragment', async () => {
+      const renderResult = await renderTestEditorWithCode(
+        projectWithFragment,
+        'await-first-dom-report',
+      )
+
+      expect(Object.keys(renderResult.getEditorState().editor.jsxMetadata)).toEqual([
+        'sb',
+        'sb/fragment',
+        'sb/fragment/child',
+      ])
+
+      FOR_TESTS_setNextGeneratedUids([
+        'fragment2',
+        'fragment2',
+        'fragment2',
+        'fragment2',
+        'fragment2',
+        'fragment2',
+      ])
+      const dragDelta = windowPoint({ x: 40, y: -25 })
+
+      const targetElement = renderResult.renderedDOM.getByTestId('child')
+      const targetElementBounds = targetElement.getBoundingClientRect()
+      const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+      const startPoint = windowPoint({ x: targetElementBounds.x + 5, y: targetElementBounds.y + 5 })
+      const endPoint = offsetPoint(startPoint, dragDelta)
+
+      await selectComponentsForTest(renderResult, [EP.fromString('sb/fragment')])
+
+      await mouseDragFromPointToPoint(canvasControlsLayer, startPoint, endPoint, {
+        modifiers: altModifier,
+      })
+
+      await renderResult.getDispatchFollowUpActionsFinished()
+
+      expect(Object.keys(renderResult.getEditorState().editor.jsxMetadata)).toEqual([
+        'sb',
+        'sb/fragment2',
+        'sb/fragment2/child',
```

Comment:
hmmm, this is suspicious to me. isn't it a problem that the uid 'child' is duplicated?

Reply:
great catch! seems like the utility function used to ensure unique UIDs (`guaranteeUniqueUids`) didn't take fragments and their children into account. Fixed in https://github.com/concrete-utopia/utopia/pull/3359/commits/65a775ac7f4697b9be24a3769162b0cafad2e9a5